### PR TITLE
Free Trials in the Form of Stripe Credit 

### PR DESCRIPTION
**Pull request by Pezmc**
## Description

Adds a free trial option to FlowForge EE, which will be used in FlowForge Cloud.

Messaging wise, this is a free trial of a project for one month (based on dollar amount configured).

Technically the trial is implemented as a Stripe account credit which is applied to the Stripe customer when the team billing is set up.

This PR is for the technical implementation, UI improvements and tweaks will be covered in a follow up, there's existing discussion over the specifics in https://github.com/flowforge/flowforge/issues/1521.

### Setup Locally

Follow the [billing setup](https://github.com/flowforge/flowforge/tree/main/docs/contribute#configuring-billing) as normal, but add the `new_customer_free_credit` entry, set to a cent value above zero.

#### User Creates Their First Team

Only addition is some subtle text above the submit button.

Note: Currently the "Create a personal team for users when they register" option is not compatible with free trials

Screenshot 2023-01-12 at 14 42 17

#### Completes Stripe Flow

No changes to this page currently.

Screenshot 2023-01-12 at 14 42 51

#### Lands on Team Page

No changes to this page currently.

Screenshot 2023-01-12 at 14 45 36

#### Creates a New Project

Screenshot 2023-01-12 at 14 46 07

#### If the user happens to visit the billing page any time they have credit

Credit is also applied for example when a paying customer suspends a project and a pro-rated refund is applied.

Screenshot 2023-01-12 at 14 45 47

## 

## Related Issue(s)

#1471 

## Checklist

 - [x] I have read the [contribution guidelines](https://github.com/flowforge/flowforge/blob/main/CONTRIBUTING.md)
 - [x] Suitable unit/system level tests have been added and they pass
 - [x] Documentation has been updated
    - [ ] Upgrade instructions
    - [x] Configuration details
    - [ ] Concepts

## Labels

 - [ ] Backport needed? -> add the `backport` label
 - [ ] Includes a DB migration? -> add the `area:migration` label

### CODE REVIEW

File: frontend/src/pages/team/Billing.vue [hunk 1]
```diff
@@ -1,4 +1,7 @@
 <template>
+    <div v-if="subscription?.customer?.balance < 0" class="ff-banner ff-banner-warning mb-3">
```
Comment (knolleary) [inline frontend/src/pages/team/Billing.vue:2]: Do we have other flavours of banner to consider? The red of warning doesn't feel appropriate for this informational message.

File: frontend/src/pages/team/createProject.vue [hunk 2]
```diff
@@ -19,6 +19,9 @@
                 <div class="mb-8 text-sm text-gray-500">
                     <template v-if="!isCopyProject">Let's get your new Node-RED project setup in no time.</template>
                 </div>
+                <div v-if="subscription?.customer?.balance < 0" class="ff-banner ff-banner-warning">
```
Comment (knolleary) [inline frontend/src/pages/team/createProject.vue:22]: As with other comment - think the red of warning is too much here.

File: frontend/src/pages/team/Billing.vue (hunk 1 above)
Comment (Pezmc) [reply frontend/src/pages/team/Billing.vue:2]: This was intended very much as a temporary banner before the #1521 improvements, but very fair point, I'll pick a lighter color.

Comment (Pezmc) [reply frontend/src/pages/team/Billing.vue:2]: Color toned down to a softer blue, Screenshot 2023-01-12 at 16 09 13, will iterate on the best way to show the credit with Joe in #1521.

**Review by knolleary — APPROVED** [review]
Two pieces of follow-up work I want to look at, but not holding up progress today.

1. I have a small concern over the ability for a user to game it to keep getting free credit by deleting their team and recreating each month. Will consider whether tracking that per-user is worth the overhead.
2. We should have audit-log entries for when credit is applied as that will help us keep an eye on what's happening (in additional to the application log).

**Comment by Pezmc** [comment]
All follow ups are being tracked in the main issue #1471.

**Pull request merged**